File: lab_2/Lab2/generator.cpp
#include <iostream>
#include <random>

std::string generateRandomBits() {
    std::random_device rd;
    std::shuffle_order_engine<std::mt19937, 2> gen(rd());

    std::string result;
    for (int i = 0; i < 128; ++i) {
        result += std::to_string(gen() % 2);
    }

    return result;
}

int main() {
    std::string randomBits = generateRandomBits();
    std::cout << randomBits << std::endl;

    return 0;
}

File: lab_2/Lab2/generator.java
import java.util.Random;

public class Generator {

    private static String generateRandomBits(Random random, int length) {
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < length; i++) {
            int bit = random.nextInt(2);
            result.append(bit);
        }
        return result.toString();
    }

    public static void main(String[] args) {
        Random random = new Random();

        String randomBits = generateRandomBits(random, 128);

        System.out.println("Random sequence: " + randomBits);
    }
}